openerp/addons/base/res/res_users.py

```diff
@@ -564,6 +576,8 @@ def has_group(self, cr, uid, group_ext_id):
                         (SELECT res_id FROM ir_model_data WHERE module=%s AND name=%s)""",
                    (uid, module, ext_id))
         return bool(cr.fetchone())
+    # for a few places explicitly clearing the has_group cache
+    has_group.clear_cache = _has_group.clear_cache
```

Comment:
Yeah the v7/v8 decorator creates some sort of merged dispatch object, which calls the correct sub-function based on the way it's called.